tests/pipeline/controlflowanalysis/test_pattern_independent_restructuring.py

```diff
@@ -4937,6 +4934,106 @@ def test_extract_return(task):
+            ),
+            BasicBlock(
+                5,
+                instructions=[
+                    Assignment(var_5, Constant(0, Integer.int32_t())),
+                    Assignment(var_7, Constant(-1, Integer.int32_t())),
+                    Assignment(arg1, Constant(0, Integer.int32_t())),
+                    Assignment(var_2, Constant(0, Integer.int32_t())),
+                ],
+            ),
+            BasicBlock(
+                6,
+                instructions=[
+                    Assignment(var_5, Constant(0, Integer.int32_t())),
+                    Assignment(var_7, Constant(-1, Integer.int32_t())),
+                    Assignment(var_2, Constant(0, Integer.int32_t())),
+                ],
+            ),
+            BasicBlock(7, instructions=[Assignment(ListOperation([]), Call(imp_function_symbol("sub_1400193a8"), []))]),
+            BasicBlock(
+                8,
+                instructions=[
+                    Assignment(ListOperation([]), Call(imp_function_symbol("scanf"), [Constant(0x804B01F), var_6])),
+                    Branch(Condition(OperationType.greater_us, [var_6, Constant(0, Integer.int32_t())])),
+                ],
+            ),
+            BasicBlock(9, instructions=[Assignment(arg1, Constant(1, Integer.int32_t()))]),
+            BasicBlock(10, instructions=[Return([arg1])]),
+        ]
+    )
+    task.graph.add_edges_from(
+        [
+            TrueCase(vertices[0], vertices[1]),
+            FalseCase(vertices[0], vertices[2]),
+            TrueCase(vertices[1], vertices[3]),
+            FalseCase(vertices[1], vertices[4]),
+            TrueCase(vertices[2], vertices[5]),
+            FalseCase(vertices[2], vertices[6]),
+            TrueCase(vertices[3], vertices[7]),
+            FalseCase(vertices[3], vertices[8]),
+            UnconditionalEdge(vertices[4], vertices[7]),
+            UnconditionalEdge(vertices[5], vertices[10]),
+            UnconditionalEdge(vertices[6], vertices[9]),
+            UnconditionalEdge(vertices[7], vertices[9]),
+            TrueCase(vertices[8], vertices[9]),
+            FalseCase(vertices[8], vertices[10]),
+            UnconditionalEdge(vertices[9], vertices[10]),
+        ]
+    )
+    PatternIndependentRestructuring().run(task)
+    assert any(isinstance(node, SwitchNode) for node in task.syntax_tree)
+    arg1_conditions = [
+        node
+        for node in task.syntax_tree.get_condition_nodes_post_order()
+        if node.condition.is_symbol and str(task.syntax_tree.condition_map[node.condition]) == "arg1 == 0x1"
+    ]
+    assert len(arg1_conditions) == 2
+    assert arg1_conditions[0] == arg1_conditions[1]
+    assert hash(arg1_conditions[0]) != hash(arg1_conditions[1])
+
+
 # logging.info(f"Abstract syntax tree of this region:")
 # for node in task._ast.topological_order():
 #     logging.info(f"Node {node}")
```

Comment:
?

Reply:
old debug part, I can remove it, but it feels at home here ;)

Reply:
removed